cmd/collector/config/config.go

```diff
@@ -154,6 +155,31 @@ func (s *PrometheusScrape) Validate() error {
 	return nil
 }
 
+type KubeletDiscovery struct {
+	Host           string `toml:"host,omitempty" comment:"Kubelet host address. Defaults to node name from NODE_NAME env var, or '127.0.0.1' if not set"`
+	Port           int    `toml:"port,omitempty" comment:"Kubelet port. Defaults to 10250"`
+	PollInterval   int    `toml:"poll-interval,omitempty" comment:"How often to poll the kubelet for pod updates (seconds). Defaults to 10"`
+	RequestTimeout int    `toml:"request-timeout,omitempty" comment:"Request timeout for kubelet API calls (seconds). Defaults to 9"`
+	TokenPath      string `toml:"token-path,omitempty" comment:"Path to service account token. Defaults to /var/run/secrets/kubernetes.io/serviceaccount/token"`
+	CAPath         string `toml:"ca-path,omitempty" comment:"Path to CA certificate for TLS verification. Defaults to /var/run/secrets/kubernetes.io/serviceaccount/ca.crt"`
+}
+
+func (k *KubeletDiscovery) Validate() error {
+	if k.Port < 0 || k.Port > 65535 {
+		return errors.New("kubelet-discovery.port must be between 0 and 65535")
+	}
+
+	if k.PollInterval != 0 && k.PollInterval < 1 {
+		return errors.New("kubelet-discovery.poll-interval must be at least 1 second")
+	}
+
+	if k.RequestTimeout != 0 && k.RequestTimeout < 1 {
+		return errors.New("kubelet-discovery.request-timeout must be at least 1 second")
+	}
```

Comment:
```suggestion
	if k.RequestTimeout != 0 && k.RequestTimeout < 1 {
		return errors.New("kubelet-discovery.request-timeout must be at least 1 second")
	}
	if k.PollInterval <= k.RequestTimeout {
		return errors.New("kubelet-discover.poll-interval must be less than kubelet-discovery.request-timeout")
	}
```

Should we ensure the `PollInterval < RequestTimeout`?